src/orange/orangeinp/IntersectRegion.hh

```diff
@@ -239,6 +239,82 @@ class EllipticalCylinder final : public IntersectRegionInterface
     real_type hh_;
 };
 
+//---------------------------------------------------------------------------//
+/*!
+ * Region formed by extruding + scaling a convex polygon along a line segment.
+ *
+ * The convex polygon is supplied as a set of points on the XY plane in
+ * clockwise order. The line segment is supplied as a length 2 array of 3D
+ * points denoting the extrusion path traversed by the polygon's origin,
+ * starting at the minimum Z and moving upward. Along the line, the size of the
+ * polygon is linearly scaled in accordance with scaling factors supplied for
+ * each line segment point.
+ *
+ * As is done in Geant4, construction is done by first applying scaling factors
+ * to the upper and lower polygons via scalar multiplication with each polygon
+ * point, then the points on the line are used to offset the upper and lower
+ * polygons.
+ *
+ * The line_segment and scaling parameters are stored as const references and
+ * must have lifetimes longer than this class.
+ */
+class ExtrudedPolygon final : public IntersectRegionInterface
+{
+  public:
+    //!@{
+    //! \name Type aliases
+    using VecReal2 = std::vector<Real2>;
+    using ArrayReal3 = celeritas::Array<Real3, 2>;
+    using ArrayReal = celeritas::Array<real_type, 2>;
+    //!@}
+
+  public:
+    // Construct from a convex polygon, line segment, and scaling factors
+    ExtrudedPolygon(VecReal2 const& polygon,
+                    ArrayReal3 const& line_segment,
+                    ArrayReal const& scaling);
```

Comment:
The `ArrayReal3` and `ArrayReal` definition is not obvious from the names: the second is really a Real2 but it's a lo/hi rather than x/y. I think maybe use an approach like `GenPrism`: combine segment + scaling into a struct, and pass two of those to the constructor: `lo` and `hi`. Then your internal data structures can use `EnumArray<Bound,  real_type>` etc.

Reply:
Sure --- but then what should I do for the accessors? If we only store EnumArrays (which I see the value in to catch indexing mistakes) then the accessors would have to reconstruct the `PolyFaces`, which seems messy. Returning an EnumArray seems bad too.